builtin/logical/ssh/path_roles.go

```diff
@@ -60,6 +60,7 @@ type sshRole struct {
 	AllowBareDomains           bool              `mapstructure:"allow_bare_domains" json:"allow_bare_domains"`
 	AllowSubdomains            bool              `mapstructure:"allow_subdomains" json:"allow_subdomains"`
 	AllowUserKeyIDs            bool              `mapstructure:"allow_user_key_ids" json:"allow_user_key_ids"`
+	AllowEmptyPrincipals       bool              `mapstructure:"allow_empty_principals" json:"allow_empty_principals"`
```

Comment:
I wonder if it makes sense to make this a `*bool` and default to some global value (on `ssh/config/certs` or something). I'm not sure how you'd unset the value in the Role though, tbh; you'd probably have to delete the role and re-create it, or accept a string (`true`, `false`, or `default`).

I'm kinda in favor of keeping parity with HashiCorp here, though, I think that's a lot of work for something few will need unless they've got bad workloads already.

Reply:
I took a slightly different approach: given `allowed_users=*` would let you get a certificate for _any_ concrete user anyways (or similarly for `allowed_hosts=*`), it isn't an issue to bypass `allow_empty_principals=false` here. This should reduce the compatibility breakage.